src/main/java/xyz/nucleoid/plasmid/game/player/JoinOffer.java

```diff
@@ -0,0 +1,120 @@
+package xyz.nucleoid.plasmid.game.player;
+
+import com.mojang.authlib.GameProfile;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.text.Text;
+import net.minecraft.util.math.Vec3d;
+import xyz.nucleoid.plasmid.game.GameSpace;
+import xyz.nucleoid.plasmid.game.GameTexts;
+import xyz.nucleoid.plasmid.game.event.GamePlayerEvents;
+
+import java.util.Collection;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+/**
+ * Represents a request for of a player or group of players to join a {@link GameSpace}.
+ * <p>
+ * This object should be used in order to construct a {@link JoinOfferResult} object to return from a listener to the
+ * {@link GamePlayerEvents#OFFER} event.
+ *
+ * @see GameSpace
+ * @see GamePlayerEvents#OFFER
+ */
+public interface JoinOffer {
+    /**
+     * @return the set of {@link GameProfile} of the players that are requesting access to this {@link GameSpace}
+     */
+    Collection<GameProfile> players();
+
+    /**
+     * @return the {@link UUID profile UUID} of the players that are requesting access to this {@link GameSpace}
+     */
+    default Collection<UUID> playerIds() {
```

Comment:
Nit: maybe a `Set`? :slightly_smiling_face:

Reply:
Ok, I wasn't initially sure if we want `List` of `Set`.